Inboveg expandsfunctions

ElsDeBie opened the pull request:
I was searching how to test the new function that I wrote in a branch? I did the build 'install and restart' and check but of course this function was not implemented and gave this error: 
By "check": 
==> devtools::check()
Updating inbodb documentation
i Loading inbodb
Error in ------get_inboveg_layer <- function(connection, survey_name,  (from get_inboveg_layer_qualifiers.R#10) : 
  object 'get_inboveg_layer' not found
Calls: <Anonymous> ... withr_with_dir -> force -> source_many -> source_one -> eval -> eval
Execution halted

Exited with status 1. 

I tested the function via a stand alone r-file likr
layer_info <- function( survey_name, .con) {
  dbGetQuery(con, glue_sql(
  "SELECT ivSurvey.Name ...

But I guess there is a way to do it in the project itself?

ElsDeBie committed:
New function for layer-qualifiers (still problem with FtActionGroupValues, to use instead of 'union' query).
Correction of example in 'get_inboveg_qualifiers.
Added webpage with more information on inboveg (English version is missing on inbo website, still has to be fixed) in the vignette.

ElsLommelen committed:
Merge branch 'master' into inboveg_expandsfunctions

ElsDeBie committed:
finished the layer function. Not tested yet

hansvancalster commented on R/get_inboveg_layer_qualifiers.R:
```diff
@@ -0,0 +1,127 @@
+#' multiple parameter, giving the name or names of the
+#' survey(s) for which you want to extract recordings information. If missing, all
+#' surveys are returned.
+--#' @param qualifier_type A character vector giving the name of qualifier type for which
+--#' you want to extract  information e.g. 'SQ' (site qualifier), 'MQ' (management qualifier).
+--#' If missing, all qualifier types are returned.
```
I think these `--` might be causing devtools::check() to error

ElsDeBie committed:
Correction in function LayerQualifier - description of function, delete of qualifier type (always LQ).
New function for layer information (cover and height) - not working yet

ElsDeBie committed:
The function Layer-qualifier and Layer-information should now work.

ElsDeBie committed:
Delete of a row that I tested as example

ElsDeBie commented:
I did the 'install and restart', and 'check': 0 errors and 0 warings.
I now start working on a connection between NBNdata en Userflora

hansvancalster commented:
@ElsDeBie after running 'check' you should see changes to namespace and *.Rd files in your git pane. Can you commit and push these changes?

hansvancalster commented on R/get_inboveg_layer_information.R:
```diff
@@ -0,0 +1,123 @@
+#' # get all layer qualifiers from MILKLIM surveys (partial matching)
+#' qualifiers_milkim <- get_inboveg_layerinfo(con, survey_name = "%MILKLIM%")
+#'
+#' # get layer qualifiers from several specific surveys
+#' qualifiers_severalsurveys <- get_inboveg_layerinfo(con, survey_name =
+#' c("MILKLIM_Heischraal2012", "NICHE Vlaanderen"), multiple = TRUE)
+#'
```
This example failed when I tried it.

hansvancalster commented on R/get_inboveg_layer_information.R:
```diff
@@ -0,0 +1,123 @@
+
+  } else {
+    sql_statement <- glue_sql(common_part,
+                              "AND ivS.Name IN {survey_name*}",
```
Is the `*` after `survey_name` necessary? I think it should be deleted and it may be the reason why the example with multiple survey names failed

hansvancalster replied:
Ignore that. The problem is missing brackets: it should be `({survey_name*})`

hansvancalster commented on R/get_inboveg_layer_qualifiers.R:
```diff
@@ -0,0 +1,123 @@
+
+  } else {
+    sql_statement <- glue_sql(common_part,
+                              "AND ivS.Name IN {survey_name*}",
```
```suggestion
                              "AND ivS.Name IN ({survey_name*})",
```

hansvancalster commented on R/get_inboveg_layer_information.R:
```diff
@@ -0,0 +1,123 @@
+
+  } else {
+    sql_statement <- glue_sql(common_part,
+                              "AND ivS.Name IN {survey_name*}",
```
```suggestion
                              "AND ivS.Name IN ({survey_name*})",
```

hansvancalster commented:
If you accept the "commit suggestions" online, don't forget to pull the changes. Can you also test if the examples indeed work as expected.

hansvancalster commented:
> I did the 'install and restart', and 'check': 0 errors and 0 warings.
> I now start working on a connection between NBNdata en Userflora

Thanks for adding these extra functions! Feel free to push more functions to this branch or start a new branch and PR (sometimes the latter is preferable if for instance development of the NBNdata function will take a long time).

hansvancalster commented on R/get_inboveg_layer_information.R:
```diff
@@ -0,0 +1,123 @@
+#' con <- connect_inbo_dbase("D0010_00_Cydonia")
+#'
+#' # get the layer information from one survey
+#' qualifiers_heischraal2012 <- get_inboveg_layerinfo(con, survey_name =
```
```suggestion
#' layerinfo_heischraal2012 <- get_inboveg_layerinfo(con, survey_name =
```

hansvancalster commented on R/get_inboveg_layer_information.R:
```diff
@@ -0,0 +1,123 @@
+#' "MILKLIM_Heischraal2012")
+#'
+#' # get all layer qualifiers from MILKLIM surveys (partial matching)
+#' qualifiers_milkim <- get_inboveg_layerinfo(con, survey_name = "%MILKLIM%")
```
```suggestion
#' layerinfo_milkim <- get_inboveg_layerinfo(con, survey_name = "%MILKLIM%")
```

hansvancalster commented on R/get_inboveg_layer_information.R:
```diff
@@ -0,0 +1,123 @@
+#' qualifiers_milkim <- get_inboveg_layerinfo(con, survey_name = "%MILKLIM%")
+#'
+#' # get layer qualifiers from several specific surveys
+#' qualifiers_severalsurveys <- get_inboveg_layerinfo(con, survey_name =
```
```suggestion
#' layerinfo_severalsurveys <- get_inboveg_layerinfo(con, survey_name =
```

hansvancalster commented on R/get_inboveg_layer_information.R:
```diff
@@ -0,0 +1,123 @@
+#' c("MILKLIM_Heischraal2012", "NICHE Vlaanderen"), multiple = TRUE)
+#'
+#' # get all layer qualifiers of all surveys
+#' allqualifiers <- get_inboveg_layerinfo(con)
```
```suggestion
#' all_layerinfo <- get_inboveg_layerinfo(con)
```

ElsLommelen commented on R/get_inboveg_layer_information.R:
```diff
@@ -0,0 +1,123 @@
+#' @examples
+#' \dontrun{
+#' library(inbodb)
+#' library(DBI)
+#' library(odbc)
```
As you are only using functions of inbodb, these packages don't have to be loaded.  I also suppose dplyr doesn't have to be loaded in this case?  (Could you please try? Restart R, only load inbodb and try the example.)  Certainly in an example, I think it is important to stick to the necessary steps and not confuse users by doing unnecessary steps.
```suggestion
```

ElsLommelen commented on R/get_inboveg_layer_qualifiers.R:
```diff
@@ -0,0 +1,123 @@
+#' @examples
+#' \dontrun{
+#' library(inbodb)
+#' library(DBI)
+#' library(odbc)
+#' library(dplyr)
```
Same remark here: as you are only using inbodb functions, loading other packages is not needed.  (I suppose we should also have a look at other functions to remove unnecessary packages in the examples.  Initially I was not aware of the fact that `dbDisconnect()` was added to inbodb, so I didn't question the fact that DBI and odbc were loaded in the examples...  dplyr on the other hand is needed in some of the examples.)
```suggestion
```

ElsLommelen reviewed (changes requested):
Just some small remarks from my part.  (Honestly, I only looked at the examples...)

And more importantly, I wanted to remind that the package website should be rebuild before merging using the code `pkgdown::build_site()`, and of course these changes should be committed and pushed.

ElsDeBie committed:
Changes after running 'check'

ElsDeBie committed:
Update R/get_inboveg_layer_qualifiers.R

Co-authored-by: Hans Van Calster <hans.vancalster@inbo.be>

ElsDeBie committed:
Update R/get_inboveg_layer_information.R

Co-authored-by: Hans Van Calster <hans.vancalster@inbo.be>

ElsDeBie committed:
Update R/get_inboveg_layer_information.R

Co-authored-by: Hans Van Calster <hans.vancalster@inbo.be>

ElsDeBie committed:
Update R/get_inboveg_layer_information.R

Co-authored-by: Hans Van Calster <hans.vancalster@inbo.be>

ElsDeBie committed:
Update R/get_inboveg_layer_information.R

Co-authored-by: Hans Van Calster <hans.vancalster@inbo.be>

ElsDeBie committed:
Update R/get_inboveg_layer_information.R

Co-authored-by: Hans Van Calster <hans.vancalster@inbo.be>

ElsDeBie committed:
Update R/get_inboveg_layer_information.R

Co-authored-by: ElsLommelen <els.lommelen@inbo.be>

ElsDeBie committed:
Update R/get_inboveg_layer_qualifiers.R

Co-authored-by: ElsLommelen <els.lommelen@inbo.be>

ElsDeBie commented:
> If you accept the "commit suggestions" online, don't forget to pull the changes. Can you also test if the examples indeed work as expected.

Sorry, I tried to find the way I can test the examples in the scripts (I know you already explained, but I don't find it back on this github space) I tought it was by install and restart. Is It? Because than I do need packages as DBI (dbGetQuery for example) to run the examples

ElsLommelen commented:
Yes, indeed, pull, then install and restart, and then run the example.
Strange that you need DBI, I just tried it here, and it runs without asking for DBI.  You didn't run the function code first, did you?  (This is the only 'easy' reason I could think of, why you would possibly be asked for DBI.  Another reason could be some installation or configuration errors, I hope this is not the case.)

hansvancalster commented:
> > If you accept the "commit suggestions" online, don't forget to pull the changes. Can you also test if the examples indeed work as expected.
> 
> Sorry, I tried to find the way I can test the examples in the scripts (I know you already explained, but I don't find it back on this github space) I tought it was by install and restart. Is It? Because than I do need packages as DBI (dbGetQuery for example) to run the examples

Strange. I just did install and restart and ran the examples (without loading extra packages - only  inbodb loaded) and it works as expected.

ElsLommelen commented:
Something offtopic related to an earlier discussion in this repo: if you want documentation to be updated when hitting `Install and Restart`, you should choose More > Configure Build Tools and then hit the button Configure and tick Install and Restart. Close twice with OK, and your done (for this repo).

@ElsDeBie Maybe you should make sure that all items but Vignettes are ticked in the list.  If not, that might cause the problem.

ElsDeBie committed:
Strange, these changes were made by Hans and I pulled the latest version of github here local. And now I got these changes in de files .. so I just push and pull again.

ElsLommelen commented:
@ElsDeBie Indeed, @hansvancalster made these changes in the R-files and asked you to pull the changes and update the documentation.  So that is what you did now, thanks!

Now just try the example once more, fingers crossed it works now!

ElsDeBie commented:
> library(inbodb)
Attaching package: ‘inbodb’
The following objects are masked _by_ ‘.GlobalEnv’:
    get_inboveg_layerinfo, get_inboveg_layerqualifier
I run:
> con <- connect_inbo_dbase("D0010_00_Cydonia")
> layerinfo_heischraal2012 <- get_inboveg_layerinfo(con, survey_name = "MILKLIM_Heischraal2012")
Error in assert_that(inherits(connection, what = "Microsoft SQL Server"),  : 
  could not find function "assert_that"

In other functions, for example > survey_info <- get_inboveg_survey(con, survey_name = "OudeLanden_1979", collect = TRUE), No error occurs. 
So the problem must be to do with : The following objects are masked _by_ ‘.GlobalEnv’:
I cleanded the 'Global environment as 'google' told me so and now IT WORKS!

ElsDeBie commented:
> Just some small remarks from my part. (Honestly, I only looked at the examples...)
> 
> And more importantly, I wanted to remind that the package website should be rebuild before merging using the code `pkgdown::build_site()`, and of course these changes should be committed and pushed.

So after these extra functions, I update the vignette and check the other functions (rmds) for unnecessary libraries. Than I will make a new branch for the next additional functions.

ElsDeBie committed:
Delete of unnecessary libraries in the function files.
Correction of names in examples (qualifier_heischraal2021 becomes layerqualifier_heischraal2021)
Expand the vignette with the 2 new layer functions

hansvancalster commented on vignettes/get_data_inboveg.Rmd:
````diff
@@ -242,13 +240,74 @@ qualifiers_severalsurveys <- get_inboveg_qualifiers(con,
 ```
 
+## Layer information
+This function `get_inboveg_layerinfo` queries the INBOVEG database for layer information on recordings  for one or more surveys.
````
```suggestion
The function `get_inboveg_layerinfo` queries the INBOVEG database for layer information on recordings  for one or more surveys.
```

hansvancalster commented on vignettes/get_data_inboveg.Rmd:
````diff
@@ -242,13 +240,74 @@ qualifiers_severalsurveys <- get_inboveg_qualifiers(con,
+```
+
+## Layer qualifiers
+This function `get_inboveg_layerqualifier`queries the INBOVEG database for layer qualifier information on recordings  for one or more surveys.
````
```suggestion
The function `get_inboveg_layerqualifier`queries the INBOVEG database for layer qualifier information on recordings  for one or more surveys.
```

ElsDeBie committed:
Update vignettes/get_data_inboveg.Rmd

Co-authored-by: Hans Van Calster <hans.vancalster@inbo.be>

ElsDeBie committed:
Update vignettes/get_data_inboveg.Rmd

Co-authored-by: Hans Van Calster <hans.vancalster@inbo.be>

hansvancalster reviewed (commented):
After:

- bumping the package version in DESCRIPTION file, 
- running devtools::document() 
- and pkgdown::build_site() (and pushing the changes), 
 
this PR should be ready to merge.

ElsDeBie commented:
found a problem in classification (the mulitple isn't finished yet). So I will first solve this and than the actionpoints @hansvancalster mentioned above.

ElsDeBie committed:
Added 'multiple' option in Classification function

ElsDeBie committed:
Added an example in the vignette using multiple = TRUE in classification-function.
Did the install/build and check of the package with no errors, warnings or notes.

ElsDeBie committed:
Run the code pkgdown::build_site() and got 2 warning messages (I think we can skip these).

== DONE ========================================================================
Warning messages:
1: The vignette title specified in \VignetteIndexEntry{} is different from the title in the YAML metadata. The former is "get_data_inboveg", and the latter is "How to retrieve data from the INBOVEG database". If that is intentional, you may set options(rmarkdown.html_vignette.check_title = FALSE) to suppress this check.
2: The vignette title specified in \VignetteIndexEntry{} is different from the title in the YAML metadata. The former is "odbc_query_problem", and the latter is "odbc query problem". If that is intentional, you may set options(rmarkdown.html_vignette.check_title = FALSE) to suppress this check.

ElsDeBie commented:
I think this branch can be merged (i did the check and build the site as suggested above by @hansvancalster and @ElsLommelen

hansvancalster reviewed (approved):
Looks good to me. @ElsLommelen do you know if the warnings about different vignette title in \VignetteIndexEntry{} should be addressed?

ElsLommelen commented:
We might want to deal with this in future, but they are warnings, and for now it builds fine...  So I can have a look at it when automating the site build, but for now we can just ignore it, I think.

ElsLommelen reviewed (approved):
Oups, I notice just now that I haven't approved this branch yet...  I'll immediately merge it.
